Microsoft.Toolkit.Uwp/Helpers/PrintHelper/PrintHelper.cs

```diff
@@ -0,0 +1,373 @@
+        /// into a page that the Windows print system can deal with.
+        /// </summary>
+        /// <param name="sender">PrintDocument</param>
+        /// <param name="e">Arguments containing the preview requested page</param>
+        private void GetPrintPreviewPage(object sender, GetPreviewPageEventArgs e)
+        {
+            PrintDocument printDoc = (PrintDocument)sender;
+            printDoc.SetPreviewPage(e.PageNumber, _printPreviewPages[e.PageNumber - 1]);
+        }
+
+        /// <summary>
+        /// This is the event handler for PrintDocument.AddPages. It provides all pages to be printed, in the form of
+        /// UIElements, to an instance of PrintDocument. PrintDocument subsequently converts the UIElements
+        /// into a pages that the Windows print system can deal with.
+        /// </summary>
+        /// <param name="sender">PrintDocument</param>
+        /// <param name="e">Add page event arguments containing a print task options reference</param>
+        private void AddPrintPages(object sender, AddPagesEventArgs e)
+        {
+            // Loop over all of the preview pages and add each one to  add each page to be printied
+            for (int i = 0; i < _printPreviewPages.Count; i++)
+            {
+                // We should have all pages ready at this point...
+                _printDocument.AddPage(_printPreviewPages[i]);
+            }
+
+            PrintDocument printDoc = (PrintDocument)sender;
+
+            // Indicate that all of the print pages have been provided
+            printDoc.AddPagesComplete();
+        }
+
+        /// <summary>
+        /// This function creates and adds one print preview page to the internal cache of print preview
+        /// pages stored in printPreviewPages.
+        /// </summary>
+        /// <param name="element">FrameworkElement that is used to represent the "printing page"</param>
+        /// <param name="printPageDescription">Printer's page description</param>
+        private void AddOnePrintPreviewPage(FrameworkElement element, PrintPageDescription printPageDescription)
+        {
+            var page = new Page();
+
+            // Set "paper" width
+            page.Width = printPageDescription.PageSize.Width;
+            page.Height = printPageDescription.PageSize.Height;
+
+            // Get the margins size
+            double marginWidth = Math.Max(printPageDescription.PageSize.Width - printPageDescription.ImageableRect.Width, printPageDescription.PageSize.Width * ApplicationContentMarginLeft * 2);
+            double marginHeight = Math.Max(printPageDescription.PageSize.Height - printPageDescription.ImageableRect.Height, printPageDescription.PageSize.Height * ApplicationContentMarginTop * 2);
+
+            // Set-up "printable area" on the "paper"
+            var stateBag = new PrintHelperStateBag();
+            stateBag.Capture(element);
+            _stateBags.Add(element, stateBag);
```

Comment:
when changing the page orientation from portrait to landscape, the preview hangs and in debug mode we get this error 

An exception of type 'System.ArgumentException' occurred in mscorlib.ni.dll but was not handled in user code

Additional information: An item with the same key has already been added. Key: Windows.UI.Xaml.Controls.StackPanel